xray/src/lib.rs

```diff
@@ -23,12 +23,27 @@ use cgmath::Point2;
 use fnv::FnvHashSet;
 use quadtree::{NodeId, Rect};
 use std::fs::File;
-use std::io::{Cursor, Read};
+use std::io::BufReader;
 use std::path::Path;
 
+pub use xray_proto_rust::proto;
+
 pub const CURRENT_VERSION: i32 = 2;
 
-#[derive(Debug)]
+pub fn read_meta_proto_from_file<P: AsRef<Path>>(filename: P) -> proto::Meta {
+    let f = File::open(&filename).expect("Could not open meta.pb");
+    let meta = protobuf::parse_from_reader::<proto::Meta>(&mut BufReader::new(f))
+        .expect("Could not parse meta.pb");
+
+    assert!(meta.version == CURRENT_VERSION,
```

Comment:
Can you add a TODO that this should return an error instead of crashing the program? This is no longer just a binary, but morphs into a library.